### DISCUSSION TITLE
Change ID Server via CLI

### DISCUSSION THREAD

**Discussion by gambitcode** [rustdesk/rustdesk#4828] [Q&A]
Is there a way to change the ID server of my RustDesk client using just the command line? I've tried editing and saving the RustDesk2.toml file under ./.config/rustdesk but even after restarting the machine or rebooting the Rustdesk application the change is not saved or applied.

**Comment by SergeyMy** [comment]
only by slightly changing the source code of the client or server...
or ..... client id is MAC_addres & 0x1FFFFFFF :)

**Reply by shvchk** [reply]
This question is not about changing ID, it's about changing ID **server**

**Comment by szymontex** [comment]
It seems that easy change ID option comes with Server Pro version which is not open source: https://rustdesk.com/pricing.html

**Reply by SergeyMy** [reply]
a significant part of the features of the Server Pro version, available when self-assembling from source, and a little bit of ingenuity

**Reply by szymontex** [reply]
How is client ID generated for now?

**Reply by SergeyMy** [reply]
> client id is MAC_addres & 0x1FFFFFFF :)

delete rustdesk config files
change the MAC address of the network card
start rustdesk

**Reply by szymontex** [reply]
Unable to change MAC address on my card.
So 2 pc's, 2 different MACs but still same ID after cloning. 
If I somehow could trigger reassign process...

**Reply by ForsakenRei** [reply]
I used my own server, as long as I remove the docker container and rustdesk on my PC then recreate/reinstall everything I will get a new ID, but it's not ideal. Kinda sad when change ID become a paid feauture though, their approach is understandable to make the project live longer.

**Reply by SergeyMy** [reply]
I repeat. changing the ID is easy - by changing just one line in the source code rendezvous_server, and compiling it yourself.
and I agree that if you are not capable of it - pay.

**Reply by akiooo45** [reply]
I want to modify rendezvous_server at compile time, where do I need to modify it?

**Reply by SergeyMy** [reply]
change NOT_SUPPORT to OK
PS everything you do you do at your own peril and risk

**Comment by akiooo45** [comment]
If I use docker to compile, may I modify env in Dockerfile
&nbsp;


我爱你
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Tianling ***@***.***&gt;; 
发送时间: 2023年7月17日(星期一) 下午4:13
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [rustdesk/rustdesk] Change ID Server via CLI (Discussion #4828)





 
change NOT_SUPPORT to OK
 PS everything you do you do at your own peril and risk
